File: day-03/workspace/exceptions-basics/src/main/java/com/targetindia/programs/ExceptionDemo.java
package com.targetindia.programs;


import lombok.extern.slf4j.Slf4j;

import java.util.Arrays;

@Slf4j
public class ExceptionDemo {

    public static void main(String[] args) {
        log.trace("start of main()");
        log.trace("you have supplied these arguments: " + Arrays.toString(args));

        if (args.length < 2) {
            log.error("two inputs required, but got {}", args.length);
            return;
        }

        var input1 = args[0];
        var input2 = args[1];
        var num = 0;
        var den = 0;

        try {
            num = Integer.parseInt(input1);
            den = Integer.parseInt(input2);
        } catch (NumberFormatException ex) {
            log.error("integer inputs were expected, but got {}", Arrays.toString(args));
            return;
        }

        if(den==0){
            log.error("cannot divide integer by zero");
            return;
        }

        var quot = num / den;
        var rem = num % den;

        log.info("dividing {} by {} results in {} as quotient and {} as remainder", num, den, quot, rem);

        log.trace("end of main()");
    }
}

File: day-13-advanced-kotlin-concepts/workspace/exception-handling/src/main/kotlin/com/targetindia/programs/ExceptionDemo.kt
package com.targetindia.programs

fun main() {
    println("start of main()")

    val num: Int
    val den = try {
        print("enter numerator: ")
        num = readln().toInt()
        print("enter denominator: ")
        readln().toInt()
    } catch (e: Exception) {
        println("only integers were expected!")
        return // exit early
    }

    try {
        val q = num / den
        val r = num % den
        println("$num / $den = $q")
        println("$num % $den = $r")
    } catch (e: ArithmeticException) {
        println("cannot divide an integer by zero")
    }

    println("end of main()")
}